nixos/lib/make-ext4-fs.nix

```diff
@@ -21,17 +22,22 @@ let
   sdClosureInfo = pkgs.buildPackages.closureInfo { rootPaths = storePaths; };
 in
 
-pkgs.stdenv.mkDerivation {
-  name = "ext4-fs.img";
+pkgs.stdenv.mkDerivation rec {
+  name = "ext4-fs.img.bz2";
 
-  nativeBuildInputs = [e2fsprogs.bin libfaketime perl lkl];
+  nativeBuildInputs = [ bzip2 e2fsprogs.bin libfaketime perl lkl ];
 
   buildCommand =
     ''
+      # foo.img.bz2
+      local name="${name}"
+      # foo.img
+      local img="''${name%.*}"
```

Comment:
The name of the temp image is not important at all. We could just as well call it `tmp.img` outright during the full build. There is no need to make it generic from the name of the derivation. `$out` is already doing all of the actually important naming for us :).